src/test/PdfTest/PdfControllerUnitTests.java

```diff
@@ -0,0 +1,733 @@
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.pdmodel.interactive.digitalsignature.PDSignature;
+import org.apache.pdfbox.pdmodel.interactive.digitalsignature.SignatureOptions;
+import org.apache.pdfbox.pdmodel.interactive.digitalsignature.visible.PDVisibleSigProperties;
+import org.apache.pdfbox.pdmodel.interactive.digitalsignature.visible.PDVisibleSignDesigner;
+import org.apache.pdfbox.pdmodel.interactive.form.PDSignatureField;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.*;
+
+public class PdfControllerUnitTests {
+
+  // Get subset of fields of type fieldType
+  private List<JSONObject> getFieldsOfType(List<JSONObject> fieldsJSON, String fieldType) {
+    List<JSONObject> newFieldsJSON = new LinkedList<>();
+    for (JSONObject field : fieldsJSON) {
+      if (field.getString("fieldType").equals(fieldType)) {
+        newFieldsJSON.add(field);
+      }
+    }
+    return newFieldsJSON;
+  }
+
+  private JSONObject createFieldJSON(
+      String fieldName, String fieldType, String[] fieldValueOptions) {
+    JSONArray fieldValueOptionsJSON = new JSONArray();
+    for (String fieldValueOption : fieldValueOptions) {
+      fieldValueOptionsJSON.put(fieldValueOption);
+    }
+    return (new JSONObject()
+        .put("fieldName", fieldName)
+        .put("fieldType", fieldType)
+        .put("fieldValueOptions", fieldValueOptionsJSON.toString())
+        .put("fieldMatchedDBName", "")
+        .put("fieldMatchedDBVariable", "")
+        .put("fieldQuestion", "Please Enter Your " + fieldName));
+  }
+
+  class JSONFieldComparator implements Comparator<JSONObject> {
+    public int compare(JSONObject a, JSONObject b) {
+      return (a.getString("fieldName").compareTo(b.getString("fieldName")));
+    }
+  }
+
+  @Test(expected = IOException.class)
+  public void getFieldInformationNullFileTest() throws IOException {
+    File pdfInput = new File("");
+    PDDocument pdfDocument = PDDocument.load(pdfInput);
+    List<JSONObject> fieldsJSON = new LinkedList<>();
+    PdfController.getFieldInformation(pdfDocument, fieldsJSON);
+    pdfDocument.close();
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void getFieldInformationNullPDFDocumentTest() {
+    PDDocument pdfDocument = null;
+    List<JSONObject> fieldsJSON = new LinkedList<>();
+    PdfController.getFieldInformation(pdfDocument, fieldsJSON);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void getFieldInformationNullFieldsJSONTest() {
+    PDDocument pdfDocument = null;
+    List<JSONObject> fieldsJSON = null;
+    PdfController.getFieldInformation(pdfDocument, fieldsJSON);
+  }
+
+  @Test
+  // NOTE: We need to fix the file so that the buttons are radio
+  public void getFieldInformationValidValuesTest1() throws IOException {
```

Comment:
whats the difference between valid values test 1 and 2?

Reply:
and 3 and 4 and so on

Reply:
Deleted them, they were there just to see what was inside these files.